codec-http/src/main/java/io/netty/handler/codec/http/HttpContentEncoder.java

```diff
@@ -319,4 +327,22 @@ public EmbeddedChannel contentEncoder() {
             return contentEncoder;
         }
     }
+
+    private static final class RequestMetaData {
```

Comment:
@naude-r Yes. To be precise, we will need to `CharSequence` constants. One for HEAD and the other for CONNECT, because we need to check the status code in case of CONNECT.